pkg/gui/merge_panel.go

```diff
@@ -135,6 +135,8 @@ func (gui *Gui) resolveConflict(selection mergeconflicts.Selection) (bool, error
 	switch selection {
 	case mergeconflicts.TOP:
 		logStr = "Picking top hunk"
+	case mergeconflicts.MIDDLE:
+		logStr = "Picking middle hunk"
 	case mergeconflicts.BOTTOM:
 		logStr = "Picking bottom hunk"
 	case mergeconflicts.BOTH:
```

Comment:
just saw this section now and it makes me wonder: what happens if you try to select 'both' hunks when there's now three hunks? Should we update the naming to say 'all' hunks?

Reply:
@jesseduffield I renamed `BOTH` to `ALL` and also changed the struct members and descriptions (without [`KeybindingMainConfig.PickBothHunks`](https://github.com/jesseduffield/lazygit/blob/487ad196a7c03d8cb3e0abc41dea42dd96db8a62/pkg/config/user_config.go#L244)).

Reply:
> just saw this section now and it makes me wonder: what happens if you try to select 'both' hunks when there's now three hunks? Should we update the naming to say 'all' hunks?

I think it would be better if "pick all" was still "pick both"... and was only picking **top and bottom**, but not the middle hunk. When solving merge conflicts with the `diff3` style, the middle hunk is the common base. But the common base has been modified in **both** the remote and local branches. It doesn't make sense to pick it together with the other changes.

In general "pick both" / "pick all" only really makes sense for added lines. Modified lines would be duplicated. And by definition lines in "base" (i.e. the middle hunk) are never "added".

Does that make any sense?

Reply:
Or to put it differently: Picking the middle hunk basically means "I want none of those changes at all.". Picking all and none at the same time... doesn't make sense :D